### PR TITLE
VACMS-10260 queries refactor: linkTeaser

### PR DESCRIPTION
**Pull request by penny-lischer**
## Description

Relates to #10260

## Screenshots
Screen Shot 2022-08-15 at 4 13 17 PM

### CODE REVIEW

File: src/lib/utils/drupalClient.ts [hunk 1]
```diff
@@ -2,19 +2,59 @@ import { DrupalClient } from 'next-drupal'
 import crossFetch from 'cross-fetch'
 import { SocksProxyAgent } from 'socks-proxy-agent'
 import { addCAs } from 'syswide-cas'
+import * as fs from 'fs'
```
Comment (timcosgrove) [inline src/lib/utils/drupalClient.ts:5]: I'd like us to remove changes to this file in favor of what we'll merge in later.

**Review by timcosgrove — CHANGES_REQUESTED** [review]

File: src/types/dataTypes/drupal/field_type.d.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export interface FieldFormattedTextWithSummary extends FieldFormattedText {
 export interface FieldLink {
   uri: string
   title: string
-  options: [string[]]
+  options: any[]
```
Comment (timcosgrove) [inline src/types/dataTypes/drupal/field_type.d.ts:26]: What is prompting this change?

Comment (penny-lischer) [reply src/types/dataTypes/drupal/field_type.d.ts:26]: If the JSON data returns an empty array i.e.options: [] the query tests fail when definition is options: [string[]]. It seems that the data coming back from the response is not of type [string[]].

**Review by timcosgrove — APPROVED** [review]

**Comment by timcosgrove** [comment]
Reworked into #94 and merged. (Rework was due to my own issues doing the merge.)

**Pull request closed without merging**